### DISCUSSION TITLE
Is there any reason to have the wiki pages disabled ?

### DISCUSSION THREAD

**Discussion by m040601** [BurntSushi/ripgrep#1681] [General] [answered]
Is there any reason you have the wiki page locked or don't wish to have it open ? 

Why not open it to the community ? 

I dont' mean this as an additional place for the "official" docs or for "selling" ripgrep. You already have that pretty well covered in the excelent "user guide", homepage etc.

Neither do I mean it as a place for developers/programmers comparing/discussing the performance/ins/outs/pros/cons of ripgrep. You already have that in full control in the repo and the issues.

I mean the wiki as place for the **end users**. The ones that are not going to use "git", or github issues. There is a big difference betwenn forking the repo, to submit a pull request to the docs, and a quick wiki edit. Some end users might do it, but the majority  is not gonna  fork a repo just for adding a small tip.

Why not have a free space, the wiki, to let users, themselves, post their snippets, use cases, pipes, gotchas, etc. If you feel that you might "loose control",  you can at least give it a shot, and later if you feel like it, close it again.

I ask because this is such a very generic usefull tool, that plugs with so many different other tools, caters to many different types of "uses" and "users".

This means it will be impossible for you to maintain or document every single use case, trick or hack this tool can be used for.

Even more, it will be impossible to even "imagine" the cool things many different minds can come up with a powerfull search tool . The kind of things that make you think: "this little oneliner/pipe/script isnt that complicated, why hadnt I thought myself about that"

Search the last 30 years of internet for grep oneliners/tutorials/use cases and you'll understand my point.

Have a look at the wiki of projects like fzf (or lf, or ranger) for example. So much value can be extracted from that community wiki.

**Comment by BurntSushi** [comment, marked as answer]
I did indeed disable it, mostly because I just don't have the time to curate it. And the wiki kind of lives on its own in GitHub's little world, in a separate repository and isn't exactly easy to bundle into the release.

I do think having a document describing tricks though would indeed be quite helpful. Perhaps a new top-level `TRICKS.md` in the root of the repo? Then we could add it to the release tarball pretty easily and there would be space for others to contribute to it.

Do you want to start it?

**Comment by m040601** [comment]
> ...having a document ...
> Perhaps a ..... in the root of the repo? 
> Do you want to start it?

No. Sorry. Perhaps I wasn't clear or I didnt explain my self.
This is precisely not what I was asking for. Not something in the repo. Not something more to control and curate.


A github wiki, lives in another world. And indeed it is in its own a git repo as well.  Although it is "owned" by you, it is separated from the project. It's not the "official" docs/scripts/bundled_examples. 


> I just don't have the time to curate it

It doesnt have to be curated, edited. It's not meant to. It's meant to grow on it's own. It's either on or off. It's meant for the project developer to relinquish control, and let "the community" put there whatever it deems as worthy.  "Put" in the sense of practical real world,  tips/tricks/use cases/workflows. Not in the sense of discussing/commenting/personal opinionating. In a  a true "bazaar" sense.

You dont have to be afraid that some might read "fake information" on the wiki. You dont have to be afraid, that someone is lazy to read the "official" docs, because of the wiki's existence.

This is clear for anyone using ripgrep.It is simply a free space, you allow the world to leave there what the world finds relevant to "say" about ripgrep.  It's not the voice of BurntSushi, it's the voice of the world. That part of the world that would like to have a "central" platform to do that, instead of spending that time Reddit/Hackernews, or other blog posts etc.
 
.

> .. wiki kind of lives on its own in GitHub's little world, in a separate repository and isn't exactly easy to bundle into the release...


Precisely ! It is not meant to be bundled later in the project. Although I'm 100% sure, you'd, later,  find stuff there to inspire you to edit/change the project's own docs/shipped stuff.

I can fully understand and respect that you might not what to have an "open" wiki on github. For what ever reason. And by the way, also thank for you for all the magnificient work done in this tool. Including the docs :-)

**Comment by BurntSushi** [comment]
> It doesnt have to be curated, edited. It's not meant to. 

The only successful wikis I know _are_ heavily curated and I just don't have the time to do that.
